cheddar/cheddar-integration-aws/src/main/java/com/clicktravel/infrastructure/persistence/aws/cloudsearch/CloudSearchEngine.java

```diff
@@ -211,12 +211,19 @@ private Object getPropertyValue(final Document document, final PropertyDescripto
 
     @Override
     public void delete(final Document document) {
-        final DocumentConfiguration documentConfiguration = getDocumentConfiguration(document.getClass());
-        final String searchDomain = documentConfigurationHolder.schemaName() + "-" + documentConfiguration.namespace();
-        final BatchDocumentUpdateRequest batchDocumentUpdateRequest = new BatchDocumentUpdateRequest(searchDomain);
-        final DocumentUpdate csDocument = new DocumentUpdate(Type.DELETE, document.getId());
-        batchDocumentUpdateRequest.withDocument(csDocument);
-        getDocumentServiceClient(searchDomain).uploadDocuments(uploadDocumentsRequest(batchDocumentUpdateRequest));
+        delete(Arrays.asList(document), document.getClass());
+    }
+    
+    @Override
+    public void delete(final Collection<? extends Document> documents, final Class<? extends Document> documentClass) {
```

Comment:
Can't you get the class from the first item and throw an exception if they are not all the same?